### DISCUSSION TITLE
Instructions for cross compiling for arm

### DISCUSSION THREAD

**Discussion by JarrettR** [diasurgical/DevilutionX#3411] [Q&A] [answered]
This is evidently possible from the platform being listed in releases. I can't find it in either the docs page on building, or in the wiki.

I currently have this:

```
wget https://zlib.net/zlib-1.2.11.tar.gz -Ozlib-1.2.11.tar.gz
tar -xzf zlib-1.2.11.tar.gz
cmake .. DCMAKE_C_COMPILER=arm-linux-gnueabihf-gcc -DCMAKE_CXX_COMPILER=arm-linux-gnueabihf-g++ -DCMAKE_TOOLCHAIN_FILE=../CMake/32bit.cmake -DZLIB_ROOT=/libzlib-1.2.11/ 
```

But am still running into this issue:

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
M_LIBRARY
    linked by target "png" in directory /home/jarrett/Documents/linux/diablo/devilutionX/build/_deps/png-src
    linked by target "png-fix-itxt" in directory /home/jarrett/Documents/linux/diablo/devilutionX/build/_deps/png-src
```

And (for the moment) have run out of time to troubleshoot.

**Comment by AJenbo** [comment]
This is the first i hear of gnueabihf, where are you seeing it mentioned?

Have you installed libpng, what disto are you on?

**Comment by AJenbo** [comment, marked as answer]
Ok, looks like you are cross compiling for armhf. Yes we support armhf, but we do the compiles natively on an ARM CPU running 32bit Debian. So the scenario you trying isn't officially supported. So this isn't considered an issue, more a question.

**Comment by JarrettR** [comment]
Yeah, you got it. That's good to know that you are compiling natively. I don't currently have that ability, so I will chip away at this in a bit.

**Reply by AJenbo** [reply]
Cool, if you find that there are changes needed feel free to send a PR and we can make it official :)

**Comment by glebm** [comment]
If you're cross-compiling, you need to provide the correct toolchain file to cmake.

This is not it: `-DCMAKE_TOOLCHAIN_FILE=../CMake/32bit.cmake`

The cmake toolchain file is usually provided by your compilation environment. For example, in buildroot, this file is located at `${SYSROOT}/usr/share/buildroot/toolchainfile.cmake`.

There is no difference between cross-compiling DevilutionX and any other CMake project, as long as you supply the correct toolchain file.
